src/client/datascience/commands/activeEditorContext.ts

```diff
@@ -185,17 +184,13 @@ export class ActiveEditorContextService implements IExtensionSingleActivationSer
             this.canInterruptInteractiveWindowKernelContext.set(false).ignoreErrors();
         }
     }
-    private onDidKernelStatusChange({ notebook }: { status: ServerStatus; notebook: INotebook }) {
-        // Ok, kernel status has changed.
-        const activeEditor = this.vscNotebook.activeNotebookEditor;
-        const activeInteractiveWindow = getActiveInteractiveWindow(this.interactiveProvider);
+    private onDidKernelStatusChange({ kernel }: { status: ServerStatus; kernel: IKernel }) {
         if (
-            activeInteractiveWindow &&
-            activeInteractiveWindow.notebookUri?.toString() === notebook.identity.toString()
+            kernel.notebookDocument.notebookType === InteractiveWindowView
         ) {
             this.updateContextOfActiveInteractiveWindowKernel();
-        } else if (activeEditor && activeEditor.document.uri.toString() === notebook.identity.toString()) {
-            this.updateContextOfActiveNotebookKernel(activeEditor);
+        } else if (kernel.notebookDocument.notebookType === JupyterNotebookView && kernel.notebookDocument === this.vscNotebook.activeNotebookEditor?.document) {
```

Comment:
Some cleanup to use IKernel
Also comparing notebook documents instead of comparing strings.